subprojects/core/src/integTest/groovy/org/gradle/api/HttpScriptPluginIntegrationSpec.groovy

```diff
@@ -65,6 +66,96 @@ class HttpScriptPluginIntegrationSpec extends AbstractIntegrationSpec {
         "https" | true
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/2891")
+    def "can apply script with URI containing a query string"() {
+        when:
+        def queryString = 'p=foo;a=blob_plain;f=bar;hb=foo/bar/foo'
```

Comment:
This a somewhat unusual query String. Is that specific to a tool that generates it?